### PR TITLE
langchain_community: Add DocumentDBVectorSearch VectorStore

### PR DESCRIPTION
**Pull request by samkhano1**
**Description:**
- Added Amazon DocumentDB Vector Search integration (HNSW index)
- Added integration tests
- Updated AWS documentation with DocumentDB Vector Search instructions
- Added notebook for DocumentDB integration with example usage

### CODE REVIEW

**Comment by ip9inderpreet** [comment]
Hi Team, When can you merge this?

**Comment by samkhano1** [comment]
Hi @eyurtsev, @hwchase17, can we please get this PR reviewed? Thank you

**Comment by andywonderboi** [comment]
+1 for DocumentDB, wanted to ask, will documentation be provided like for other vector stores?

**Comment by samkhano1** [comment]
Hi @baskaryan, @efriis, can we please get this PR reviewed? Thank you

**Comment by samkhano1** [comment]
Quick bump on the above. @eyurtsev, @hwchase17, @baskaryan, @efriis, @ip9inderpreet . Thank you

File: libs/langchain/langchain/vectorstores/__init__.py
Comment (baskaryan) [inline libs/langchain/langchain/vectorstores/__init__.py:1]: can we remove any changes to libs/langchain, these imports are only for backwards compatibility

**Comment by baskaryan** [comment]
one comment, otherwise lgtm. cc @3coins

File: libs/langchain/langchain/vectorstores/__init__.py
Comment (samkhano1) [reply libs/langchain/langchain/vectorstores/__init__.py:1]: Thanks for the feedback. Fixed

**Review by baskaryan — APPROVED** [review]

**Comment by samkhano1** [comment]
Made some minor adjustments to get the build to pass. Also, added some HNSW params. @baskaryan please take another look. Thanks

**Pull request merged**